File: Data Structures and Algorithms/C/Linked List/doublycircular.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node {
    int data;
    struct Node* next;
    struct Node* prev;

} Node;

int main(void) {
    Node* node1 = (Node*) malloc(sizeof(Node));
    Node* node2 = (Node*) malloc(sizeof(Node));
    Node* node3 = (Node*) malloc(sizeof(Node));
    Node* node4 = (Node*) malloc(sizeof(Node));

    node1->data = 3;
    node2->data = 5;
    node3->data = 13;
    node4->data = 2;

    node1->next = node2;
    node1->prev = node4;

    node2->next = node3;
    node2->prev = node1;

    node3->next = node4;
    node3->prev = node2;

    node4->next = node1;
    node4->prev = node3;

    printf("\nTraversing forward: \n");
    Node* currentNode = node1;
    Node* startingNode = node1;
    printf("%d -> ", currentNode->data);
    currentNode = currentNode->next;

    while (currentNode != startingNode) {
        printf("%d -> ", currentNode->data);
        currentNode = currentNode->next;
    }
    printf("...\n"); // List loops back

    printf("\nTraversing backward: \n");
    currentNode = node4;
    startingNode = node4;
    printf("%d -> ", currentNode->data);
    currentNode = currentNode-> prev;
    
    while (currentNode != startingNode) {
        printf("%d -> ", currentNode->data);
        currentNode = currentNode->prev;
    }
    printf("...\n");  // Indicating the list loops back

    free(node1);
    free(node2);
    free(node3);
    free(node4);

    return 0;
}

File: Data Structures and Algorithms/Java/Linked Lists/DoublyCircular.java
public class DoublyCircular{
    static class Node {
        int data;
        Node next;
        Node Prev;

        Node(int data){
            this.data = data;
            this.next = null;
            this.prev = null;
        }
    }

    public static void main(String args[]){
        Node node1 = new Node(3);
        Node node2 = new Node(5);
        Node node3 = new Node(13);
        Node node4 = new Node(2);

        node1.next = node2;
        node1.prev = node4;

        node2.next = node3;
        node2.prev = node1;

        node3.next = node4;
        node3.prev = node2;

        node4.next = node1;
        node4.prev = node3;

        currentNode = node1;
        startingNode = node1;
        System.out.println("\nTraversing forward:");
        System.out.print(currentNode.data + " -> ");
        currentNode = currentNode.next;

        while (currentNode != startingNode){
            System.out.print(currentNode.data + " -> ");
            currentNode = currentNode.next;
        }

        System.out.println("..."); // List loops back

        System.out.println("\nTraversing backward:");
        currentNode = node4;
        startingNode = node4;
        System.out.print(currentNode.prev + " -> ");
        currentNode = currentNode.prev;

        while (currentNode != startingNode){
            System.out.print(currentNode.data + " -> ");
            currentNode = currentNode.prev;
        }
        System.out.println("...");
    }
}